src/functions/module_math.ts

```diff
@@ -1357,6 +1360,70 @@ export const SQRT = {
   isExported: true,
 } satisfies AddFunctionDescription;
 
+// -----------------------------------------------------------------------------
+// SUBTOTAL
+// -----------------------------------------------------------------------------
+
+export const SUBTOTAL = {
+  description: _t(
+    "Returns a subtotal for a vertical range of cells using a specified aggregation function."
+  ),
+  args: [
+    arg("function_code (number)", _t("The function to use in subtotal aggregation.")),
+    arg("ref1 (meta, range<meta>)", _t("The range or reference for which you want the subtotal.")),
+    arg(
+      "ref2 (meta, range<meta>, repeating)",
+      _t("Additional ranges or references for which you want the subtotal.")
+    ),
+  ],
+  compute: function (
+    functionCode: Maybe<FunctionResultObject>,
+    ...refs: Matrix<{ value: string }>[]
+  ) {
+    let code = toInteger(functionCode, this.locale);
+    let acceptHiddenCells = true;
+    if (code > 100) {
+      code -= 100;
+      acceptHiddenCells = false;
+    }
+    if (code < 1 || code > 11) {
+      return new EvaluationError(
+        _t("The function code (%s) must be between 1 to 11 or 101 to 111.", code)
+      );
+    }
+
+    const evaluatedCellToKeep: EvaluatedCell[] = [];
+
+    for (const ref of refs) {
+      const ref0 = ref[0][0];
+      const sheetName = splitReference(ref0.value).sheetName;
+      const sheetId = sheetName ? this.getters.getSheetIdByName(sheetName) : this.__originSheetId;
+
+      if (!sheetId) continue;
+
+      const hiddenRows = acceptHiddenCells ? [] : this.getters.getHiddenRowsGroups(sheetId).flat();
+
+      const { top, left } = toZone(ref0.value);
+      const right = left + ref.length - 1;
+      const bottom = top + ref[0].length - 1;
+
+      for (let row = top; row <= bottom; row++) {
+        if (this.getters.isRowFiltered(sheetId, row)) continue;
```

Comment:
Now, all commands related to filters need to invalidate the evaluation

Reply:
done in the commit "TOFIXUP"